### PR TITLE
Adds a fixed-wing aircraft to the competition

### PR DESCRIPTION
**Pull request by arjo129**
This PR adds a fixed wing aircraft to the simulator. The aircraft itself is taken from gazebo classic's old model repository.
To launch run:
```
ros2 launch mbzirc_ign spawn.launch.py name:=zephyr world:=simple_demo model:=mbzirc_fixed_wing type:=uav x:=1 y:=2 z:=2
```

This should spawn the vehicle with relevant plugins.
image

To launch the air craft run:
```
 ros2 topic pub /zephyr/cmd/motor_speed std_msgs/msg/Float64 data:\ 1000.0
```

flight_compressed

Additionally you may use the `/zephyr/cmd/left_flap` and `/zephyr/cmd/right_flap` to control the flaps.

## TODO
- [x] Create ROS bridge for control surfaces and thruster
- [ ] Characterize aircraft flight
- [x] Skids to make it possible to take off.
- [ ] Unit tests

### CODE REVIEW

**Comment by arjo129** [comment]
There's weird collision behavior with the height-map our plane is capable of ninja tunneling... I think its worth checking this in first and then revisiting this later. Currently uncontrolled flight doesn't work very well but then again, uncontrolled flight of fixed wings will inevitably lead to crashes.

**Comment by iche033** [comment]
I got a couple of errors on spawn and the patch fixes them for me. It also limits the flap joint vel otherwise it was flipping the model over with large torque I think.

<details><summary>patch</summary>

```c++
diff --git a/mbzirc_ign/models/mbzirc_fixed_wing_base/meshes/wing.dae b/mbzirc_ign/models/mbzirc_fixed_wing_base/meshes/wing.dae
index 902572c..fc3f453 100644
--- a/mbzirc_ign/models/mbzirc_fixed_wing_base/meshes/wing.dae
+++ b/mbzirc_ign/models/mbzirc_fixed_wing_base/meshes/wing.dae
@@ -504,7 +504,7 @@
   </library_lights>
   <library_images>
     <image id="Wing_png">
-      <init_from>wing.png</init_from>
+      <init_from>../materials/textures/wing.png</init_from>
     </image>
   </library_images>
   <library_visual_scenes>
diff --git a/mbzirc_ign/models/mbzirc_fixed_wing_base/model.sdf b/mbzirc_ign/models/mbzirc_fixed_wing_base/model.sdf
index cd82467..4eef6ee 100644
--- a/mbzirc_ign/models/mbzirc_fixed_wing_base/model.sdf
+++ b/mbzirc_ign/models/mbzirc_fixed_wing_base/model.sdf
@@ -225,6 +225,8 @@
         <limit>
           <lower>-0.524</lower>
           <upper>0.524</upper>
+          <effort>10</effort>
+          <velocity>1.57</velocity>
         </limit>
         <dynamics>
           <damping>0.1</damping>
@@ -246,6 +248,8 @@
         <limit>
           <lower>-0.524</lower>
           <upper>0.524</upper>
+          <effort>10</effort>
+          <velocity>1.57</velocity>
         </limit>
         <dynamics>
           <damping>0.1</damping>
```
</details>

**Comment by iche033** [comment]
> There's weird collision behavior with the height-map our plane is capable of ninja tunneling... I think its worth checking this in first and then revisiting this later.

yeah the heightmap does not have collisions. That's fine and since the competition area in the new coast environment will just be open water.

File: mbzirc_ign/models/mbzirc_fixed_wing/model.sdf.erb [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+      filename="ignition-gazebo-joint-position-controller-system"
+      name="ignition::gazebo::systems::JointPositionController">
+      <joint_name>flap_right_joint</joint_name>
+      <use_velocity_commands>true</use_velocity_commands>
+      <topic>/model/<%= $model_name%>/joint/right_flap_joint/cmd_pos</topic>
+    </plugin>
+    <plugin
+      filename="ignition-gazebo-joint-controller-system"
+      name="ignition::gazebo::systems::JointController">
+      <joint_name>propeller_joint</joint_name>
```
Comment (iche033) [inline mbzirc_ign/models/mbzirc_fixed_wing/model.sdf.erb:57]: should we limit the max cmd for the propeller joint so it moves at reasonable fixed wing flight speed? I think the downside is that it's harder to launch the UAV?

File: mbzirc_ign/launch/spawn.launch.py [hunk 2]
```diff
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from http.server import executable
```
[the remark this answers is not in the document]
Comment (arjo129) [reply mbzirc_ign/launch/spawn.launch.py:15]: No. Whoops, wonder how it got there.

File: mbzirc_ign/models/mbzirc_fixed_wing/model.sdf.erb (hunk 1 above)
Comment (arjo129) [reply mbzirc_ign/models/mbzirc_fixed_wing/model.sdf.erb:57]: I'm actually not sure what a "Good" maximum is although I agree we should limit the speed.

Comment (iche033) [reply mbzirc_ign/models/mbzirc_fixed_wing/model.sdf.erb:36]: I found a note here that says 838 max rpm: https://github.com/osrf/uctf/blob/master/models/zephyr_delta_wing_ardupilot_demo/delta_wing.sdf#L137 but don't know the resulting speed is. Let's come back to this when we have more info on the types of fixed wing UAVs that will be used

**Review by iche033 — APPROVED** [review]

**Pull request merged**